Assets/MixedRealityToolkit.Tools/MSBuild/Scripts/Utilities.cs

```diff
@@ -23,15 +44,25 @@ public static class Utilities
         private const string AssetsFolderName = "Assets";
         private const string PackagesFolderName = "Packages";
         private const string MSBuildFolderName = "MSBuild";
-
-        public static readonly string MSBuildOutputFolder = GetNormalizedPath(Application.dataPath.Replace("Assets", MSBuildFolderName), true);
         public const string PackagesCopyFolderName = "PackagesCopy";
 
-        private static readonly string assetsPath;
+        private const string BuiltInPackagesRelativePath = @"Data\Resources\PackageManager\BuiltInPackages";
+
+        public static string MSBuildOutputFolder { get; } = GetNormalizedPath(Application.dataPath.Replace("Assets", MSBuildFolderName), true);
+        public static string PackagesCopyPath { get; } = Path.Combine(MSBuildOutputFolder, PackagesCopyFolderName);
+        public const string MetaFileGuidRegex = @"guid:\s*([0-9a-fA-F]{32})";
+
+        private static readonly string packagesPath;
+
+        public static string AssetPath { get; }
+
+        public static string BuiltInPackagesPath { get; }
 
         static Utilities()
         {
-            assetsPath = Path.GetFullPath(Application.dataPath);
+            AssetPath = Path.GetFullPath(Application.dataPath);
+            packagesPath = Path.GetFullPath(Application.dataPath.Replace("Assets", "Packages"));
```

Comment:
>Application.dataPath.Replace("Assets", "Packages") [](start = 44, length = 50)

Be wary when the path leading up the application path can also contain the "Assets" word -  

[https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5452/files](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5452/files)

Otherwise a path like: d:/Assets/MyProject/Assets would turn into:

d:/Packages/MyProject/Packages

Where only the last one is the one we really care to replace

Same about any usage of "replace" elsewhere - I *THINK* that Assets is probably the most general so probably the one to really watch out for.

Reply:
Agreed, will update.

---
In reply to: [310757064](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5516#discussion_r310757064) [](ancestors = 310757064)